R/plot-build.R

```diff
@@ -508,19 +508,29 @@ table_add_legends <- function(table, legends, theme) {
   table <- gtable_add_rows(table, spacing$top, pos = 0)
   table <- gtable_add_rows(table, heights$top, pos = 0)
   table <- gtable_add_grob(
-    table, legends$top, clip = "off",
+    table, legends$top %||% zeroGrob(), clip = "off",
     t = 1, b = 1, l = place$l, r = place$r,
     name = "guide-box-top"
   )
 
   # Add manual legend
   place <- find_panel(table)
-  table <- gtable_add_grob(
-    table, legends$inside, clip = "off",
-    t = place$t, b = place$b, l = place$l, r = place$r,
-    name = "guide-box-inside"
-  )
-
+  inside_legends <- legends[startsWith(names(legends), "inside")]
+  if (length(inside_legends)) {
+    for (i in seq_along(inside_legends)) {
+      table <- gtable_add_grob(
+        table, inside_legends[[i]], clip = "off",
+        t = place$t, b = place$b, l = place$l, r = place$r,
+        name = paste("guide-box-inside", i, sep = "-")
+      )
+    }
+  } else { # to be consistent with original gtable layout
+    table <- gtable_add_grob(
+      table, zeroGrob(), clip = "off",
+      t = place$t, b = place$b, l = place$l, r = place$r,
+      name = "guide-box-inside"
+    )
+  }
```

Comment:
I think it should be possible to avoid this by combining all packaged inside guides into a single grob.

Reply:
fixed! Thanks for your suggestions!